twitchAPI/oauth.py

```diff
@@ -372,11 +375,14 @@ async def authenticate(self,
             # wait for the server to start up
             while not self._server_running:
                 await asyncio.sleep(0.01)
-            # open in browser
-            browser = webbrowser.get(browser_name)
-            browser.open(self._build_auth_url(), new=browser_new)
-            while self._user_token is None:
-                await asyncio.sleep(0.01)
+            if use_browser:
+                # open in browser
+                browser = webbrowser.get(browser_name)
+                browser.open(self._build_auth_url(), new=browser_new)
+                while self._user_token is None:
```

Comment:
You moved the check (the while-loop), if we received an authorization code from Twitch, in the block, which is only executed, when a browser is opened. So if no browser is automatically opened the code will proceed if the auth code is present and will then raise an TwitchAPIException.
Moving the  while loop outside the check for opening a browser should fix this.

Reply:
Thanks for this, yup I had included that higher than intended!